blocks/ai-text/src/app/generate-text.tsx

```diff
@@ -61,61 +52,64 @@ export const GenerateText = ({ blockEntity }: { blockEntity: RootEntity }) => {
     },
   } = blockEntity;
 
-  const onSubmit = useCallback(
-    async (event: FormEvent) => {
-      event.preventDefault();
-
-      if (loading || !promptText.trim()) {
-        return;
+  const onSubmit = useCallback(async () => {
+    if (loading || !promptText.trim()) {
+      return;
+    }
+
+    setError(false);
+    setLoading(true);
+
+    const isTurbo = model === "gpt-3.5-turbo";
+    const { data, errors } = await (isTurbo
+      ? serviceModule.openaiCompleteChat({
+          data: {
+            max_tokens: 4000 - promptText.length,
+            messages: [{ role: "user", content: promptText }],
+            model: "gpt-3.5-turbo",
+          },
+        })
+      : serviceModule.openaiCompleteText({
+          data: {
+            max_tokens:
+              (model === "text-davinci-003" ? 4000 : 2000) - promptText.length,
+            model,
+            prompt: promptText,
+          },
+        }));
+
+    const choice = data?.choices[0];
+
+    let textResponse: string | undefined;
+    if (choice) {
+      if ("message" in choice) {
+        textResponse = choice.message?.content;
+      } else if ("text" in choice) {
+        textResponse = choice.text;
       }
+    }
 
-      setErrorMessage("");
-      setLoading(true);
+    if (errors || !textResponse) {
+      setError(true);
+      setLoading(false);
+      return;
+    }
 
-      const isTurbo = model === "gpt-3.5-turbo";
-      const { data, errors } = await (isTurbo
-        ? serviceModule.openaiCompleteChat({
-            data: {
-              max_tokens: 4000 - promptText.length,
-              messages: [{ role: "user", content: promptText }],
-              model: "gpt-3.5-turbo",
-            },
-          })
-        : serviceModule.openaiCompleteText({
-            data: {
-              max_tokens:
-                (model === "text-davinci-003" ? 4000 : 2000) -
-                promptText.length,
-              model,
-              prompt: promptText,
-            },
-          }));
+    setGeneratedText(textResponse.replace(/^\n\n/, ""));
```

Comment:
Can we just use `trim()` here instead?
```suggestion
    setGeneratedText(textResponse.trim());
```